### PR TITLE
Touch Bug #1416256: String tiers resources util

### PR DESCRIPTION
**Pull request by phlax**
Adds a Resources util for adding/removing resources from tags according to project and project set up

it does the following

- search for linked resources
- search for not linked resources
- un/link resources

### CODE REVIEW

File: pontoon/tags/utils/resources.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
+
+from pontoon.base.utils import glob_to_regex
+
+from pontoon.tags.exceptions import InvalidProjectError
+
+from .base import TagsDataTool
+
+
+class TagsResourcesTool(TagsDataTool):
```
Comment (mathjazz) [inline pontoon/tags/utils/resources.py]: A short comment explaining the purpose of this class would be helpful.

File: pontoon/tags/utils/resources.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    filter_methods = ('locales', 'projects', 'slug', 'path')
+
+    @property
+    def data_manager(self):
+        return self.resource_manager
+
+    def link(self, tag, glob=None, resources=None):
```
Comment (mathjazz) [inline pontoon/tags/utils/resources.py]: Please add a quick comment explaining both scenarious.

File: pontoon/tags/utils/resources.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        return self.locale_manager.filter(
+            pk__in=self.get(slug).values_list(
+                "translatedresources__locale").distinct())
+
+    def get_linked_projects(self, slug):
+        return self.project_manager.filter(
+            pk__in=self.get(slug).values_list(
+                "project").distinct())
+
+    def unlink(self, tag, glob=None, resources=None):
```
Comment (mathjazz) [inline pontoon/tags/utils/resources.py]: Maybe move up, next to the `link()` method?

And also, comment. :)

File: pontoon/tags/utils/resources.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        else:
+            resources = self.resource_manager
+        if self.projects:
+            resources = resources.filter(project__in=self.projects)
+        return resources.filter(path__regex=glob_to_regex(glob))
+
+    def get(self, tag):
+        return self.filtered_data.filter(tag__slug=tag).distinct()
+
+    def get_linkable_resources(self, slug):
```
Comment (mathjazz) [inline pontoon/tags/utils/resources.py]: Please add a comment explaining linkable means "not linked yet".

File: pontoon/tags/utils/resources.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+    def get_linked_locales(self, slug):
+        return self.locale_manager.filter(
+            pk__in=self.get(slug).values_list(
+                "translatedresources__locale").distinct())
+
+    def get_linked_projects(self, slug):
+        return self.project_manager.filter(
+            pk__in=self.get(slug).values_list(
+                "project").distinct())
```
Comment (mathjazz) [inline pontoon/tags/utils/resources.py]: There's an inconsistency between `get_linked_resources` vs. `get_linked_locales` and `get_linked_projects` in that the former returns `values` and the latter two return `values_list`.

File: pontoon/tags/utils/resources.py (hunk 3 above)
Comment (phlax) [reply pontoon/tags/utils/resources.py]: ill move link down - they were supposed to be in alpha order, but link seems to have floated up

File: pontoon/tags/utils/resources.py (hunk 5 above)
Comment (phlax) [reply pontoon/tags/utils/resources.py]: ill update the the latter 2 to return values

Comment (phlax) [reply pontoon/tags/utils/resources.py]: in the end i removed the latter 2 methods - they were part of the site-wide tags implementation, and are not required/used - so better remove, at least for now

**Pull request merged**